WordPress/Classes/ViewRelated/Tools/SettingsTextViewController.h

```diff
@@ -1,13 +1,38 @@
 #import <UIKit/UIKit.h>
 
+
+// Typedef's
+typedef NS_ENUM(NSInteger, SettingsTextModes) {
+    SettingsTextModesText,
+    SettingsTextModesEmail,
+    SettingsTextModesPassword
+};
+
+typedef void (^SettingsTextChanged)(NSString *);
+
+/// Reusable component that renders a UITextField + Hint onscreen. Useful for Text / Password / Email data entry.
+///
 @interface SettingsTextViewController : UITableViewController
 
-@property (nonatomic, copy) void(^onValueChanged)(NSString *);
-@property (nonatomic, copy) void(^onCancel)();
+/// Block to be executed on dismiss, if the value was effectively updated.
+///
+@property (nonatomic, copy) SettingsTextChanged onValueChanged;
+
+/// Sets the Text Input Mode:
+///
+/// - SettingsTextModesText: Default mode
+/// - SettingsTextModesEmail: Will perform Email validation before hitting the callback.
+/// - SettingsTextModesPassword: Secure Text Entry is enabled.
+///
+@property (nonatomic, assign) SettingsTextModes mode;
```

Comment:
This is already much better, although I'd still like to see the logic moved outside the VC and tested. It's not too complicated for now, so it's up to you, but if we plan to add URL/password validation, I think we should do it.

I know it's not much logic now, but I look at this:

```
- (BOOL)textPassesValidation
{
    BOOL isEmail = (self.mode == SettingsTextModesEmail);
    return (isEmail == false || (isEmail && self.textField.text.isValidEmail));
}
```

And I wonder how that `if` evolves if we start validating for other things.